src/Server/modules/USPlacesManager.js

```diff
@@ -0,0 +1,331 @@
+const dataClassesModule = require('../modules/DataClasses.js');
+const readCSVModule = require('../modules/ReadCSV.js');
+const writeCSVModule = require('../modules/WriteCSV.js');
+
+var allUSPlaces = [];
+
+function InitializeAllUSPlace(){
+    allUSPlaces = readCSVModule.LoadUSData();
+}// read allUSdata and put into array
+
+function AddUSData(county, state, date, tod, number) {
+    
+    if(allUSPlaces == undefined){
+        InitializeAllUSPlace();
+    }
+
+	var errormsg = 'no error';
+	if (tod == 1) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numConfirmed) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						//return result;
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].currentNumConfirmed[temp_date] = number;
```

Comment:
allUSPlaces[i].currentNumConfirmed[temp_date] = number;
I think this line causes an error. Did you test it?